om4labs/om4common.py

```diff
@@ -168,3 +197,159 @@ def compute_area_regular_grid(ds, Rearth=6378e3):
     dy = dy2d
     area = dx * dy
     return area
+
+
+def horizontal_grid(dictArgs=None, point_type="t", verbose=False, output_type="xarray"):
+
+    point_type = point_type.upper()
+
+    if dictArgs is not None:
+        verbose = dictArgs["verbose"]
+        basin_file = dictArgs["basin"]
+    else:
+        basin_file = None
+
+    if dictArgs is None:
+        x = np.arange(0.5, 360.5, 1.0)
+        y = np.arange(-89.5, 90.5, 1.0)
+        area = standard_grid_cell_area(y, x)
+        geolon, geolat = np.meshgrid(x, y)
+        geolon = xr.DataArray(geolon, dims=("y", "x"), coords={"y": y, "x": x})
+        geolat = xr.DataArray(geolat, dims=("y", "x"), coords={"y": y, "x": x})
+        area = xr.DataArray(area, dims=("y", "x"), coords={"y": y, "x": x})
+        nominal_x = geolon[geolon.dims[-1]]
+        nominal_y = geolat[geolat.dims[-2]]
+
+    elif dictArgs["hgrid"] is not None:
+        if verbose is True:
+            print("Using optional hgrid file for horizontal grid.")
+        ds = xr.open_dataset(dictArgs["hgrid"])
+        geolat = subsample_supergrid(ds, "y", point_type)
+        geolon = subsample_supergrid(ds, "x", point_type)
+        area = sum_on_supergrid(ds, "area", point_type)
+
+    elif dictArgs["static"] is not None:
+        if verbose is True:
+            print("Using optional static file for horizontal grid.")
+
+        ds = xr.open_dataset(dictArgs["static"])
+
+        if point_type == "T":
+            geolat = ds["geolat"]
+            geolon = ds["geolon"]
+            area = ds["areacello"]
+
+        elif point_type == "U":
+            geolat = ds["geolat_u"]
+            geolon = ds["geolon_u"]
+            area = ds["areacello_cu"]
+
+        elif point_type == "V":
+            geolat = ds["geolat_v"]
+            geolon = ds["geolon_v"]
+            area = ds["areacello_cv"]
+
+        else:
+            raise ValueError("Unknown point type. Must be T, U, or V")
+
+    elif dictArgs["gridspec"] is not None:
+        if verbose is True:
+            print("Using optional gridspec tar file for horizontal grid.")
+        tar = tf.open(dictArgs["gridspec"])
+        ds = extract_from_tar(tar, "ocean_hgrid.nc")
+        geolat = subsample_supergrid(ds, "y", point_type)  # , outputgrid='symetric')
```

Comment:
you could write a small function `grid_from_supergrid` to avoid repeting the same 3 lines